src/Storages/Freeze.cpp

```diff
@@ -179,8 +184,13 @@ PartitionCommandsResultInfo Unfreezer::unfreezePartitionsFromTableDirectory(Merg
         for (auto it = disk->iterateDirectory(table_directory); it->isValid(); it->next())
         {
             const auto & partition_directory = it->name();
+            
+            int count_underscores = std::count_if(partition_directory.begin(), partition_directory.end(), []( char c ){return c =='_';});
+            if ((format_version.has_value() && format_version == 0) || count_underscores == 4) {
+                throw Exception(ErrorCodes::INVALID_PARTITION_VALUE, "Can not complete unfreeze query because part directory name is obsolete: " + partition_directory);
```

Comment:
`throw Exception(ErrorCodes::INVALID_PARTITION_VALUE, "Can not complete unfreeze query because part directory name is obsolete: {}", partition_directory);`

Reply:
Also exception message is not clear, what do you think about something like "because table was created with deprecated syntax and has old partitioning format"?